node-gui/backend/src/messages.rs

```diff
@@ -164,7 +162,7 @@ pub struct SendDelegateToAddressRequest {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct TransactionInfo {
     pub wallet_id: WalletId,
-    pub tx: SignedTransaction,
+    pub tx: HexEncoded<SignedTransaction>,
 }
```

Comment:
I don't really like when `HexEncoded` is used like that. IMO its usage is only justified if you use it ad-hoc, like e.g. when you need to return a `SignedTransaction` from a function and you need the return type to be serializable, you can return `HexEncoded<SignedTransaction>` instead. But making it a field of a struct looks excessive and it is very intrusive. I suggest that we use serde(with) instead.

For this you would define a module somewhere in `serialization`:
```
pub mod hex_encoded_serialization {
    use serde::Deserialize as _;
    use serialization::{
        hex::{HexDecode, HexEncode},
        Decode, Encode,
    };

    pub fn serialize<T: Encode, S: serde::Serializer>(t: &T, serializer: S) -> Result<S::Ok, S::Error> {
        let hex = t.hex_encode();
        serializer.serialize_str(&hex)
    }

    pub fn deserialize<'de, T: Decode, D: serde::Deserializer<'de>>(
        deserializer: D,
    ) -> Result<T, D::Error> {
        let hex = String::deserialize(deserializer)?;
        let value = T::hex_decode_all(hex).map_err(serde::de::Error::custom)?;
        Ok(value)
    }
}
```
And then you can write:
```
#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
pub struct TransactionInfo {
    pub wallet_id: WalletId,
    #[serde(with = "hex_encoded_serialization")]
    pub tx: SignedTransaction,
}
```

Reply:
done